### PR TITLE
chore(sdk): adding bucket events sdk spec test for the bucket event

### PR DESCRIPTION
**Pull request by tsuf239**
fixes: https://github.com/winglang/wing/issues/2452

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Monada Contribution License](https://docs.winglang.io/terms-and-policies/contribution-license.html)*.

### CODE REVIEW

File: examples/tests/sdk_tests/bucket/events.w [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        i = i + 1;
+        if this.assertion() {
+          assert(this.assertion());
+          return;
+        } 
+        this.sleep(1000 * 10);
+      }
+      assert(this.assertion());
+    }
+}
```
Comment (Chriscbr) [inline examples/tests/sdk_tests/bucket/events.w:51]: nit: fix indentation
```suggestion
    inflight testAssertion ()  {
        let var i = 0;
        // waiting for up to 2 minutes
        while i < 12 {
          i = i + 1;
          if this.assertion() {
            assert(this.assertion());
            return;
          } 
          this.sleep(1000 * 10);
        }
        assert(this.assertion());
      }
    }
```

File: examples/tests/sdk_tests/function/timeout.w [hunk 2]
```diff
@@ -8,14 +8,13 @@ let c = new cloud.Counter();
...
-  extern "../external/sleep.js" inflight sleep(ms: num);
+  extern "../external/sleep.js" static inflight sleep(ms: num);
 }
 
-let helper = new Helper();
 
 let f = new cloud.Function(inflight () => {
   // call sleep
-  helper.sleep(1500);
+  Helper.sleep(1500);
```
Comment (Chriscbr) [inline examples/tests/sdk_tests/function/timeout.w:17]: 🙌

File: libs/wingsdk/test/target-sim/__snapshots__/api.test.ts.snap [hunk 3]
```diff
@@ -1055,7 +1055,7 @@ exports[`api supports every method type 1`] = `
   "Invoke (payload={\\"headers\\":{\\"host\\":\\"127.0.0.1:<port>\\",\\"connection\\":\\"keep-alive\\",\\"accept\\":\\"*/*\\",\\"accept-language\\":\\"*\\",\\"sec-fetch-mode\\":\\"cors\\",\\"user-agent\\":\\"undici\\",\\"accept-encoding\\":\\"gzip, deflate\\"},\\"body\\":{},\\"method\\":\\"DELETE\\",\\"path\\":\\"/hello\\",\\"query\\":{},\\"vars\\":{}}).",
   "DELETE /hello - 200.",
   "Processing \\"GET /hello\\" (body={}, params={}).",
-  "Invoke (payload={\\"headers\\":{\\"host\\":\\"127.0.0.1:<port>\\",\\"connection\\":\\"keep-alive\\",\\"accept\\":\\"*/*\\",\\"accept-language\\":\\"*\\",\\"sec-fetch-mode\\":\\"cors\\",\\"user-agent\\":\\"undici\\",\\"accept-encoding\\":\\"gzip, deflate\\"},\\"body\\":{},\\"method\\":\\"HEAD\\",\\"path\\":\\"/hello\\",\\"query\\":{},\\"vars\\":{}}).",
+  "Invoke (payload={\\"headers\\":{\\"host\\":\\"127.0.0.1:<port>\\",\\"connection\\":\\"close\\",\\"accept\\":\\"*/*\\",\\"accept-language\\":\\"*\\",\\"sec-fetch-mode\\":\\"cors\\",\\"user-agent\\":\\"undici\\",\\"accept-encoding\\":\\"gzip, deflate\\"},\\"body\\":{},\\"method\\":\\"HEAD\\",\\"path\\":\\"/hello\\",\\"query\\":{},\\"vars\\":{}}).",
```
Comment (Chriscbr) [inline libs/wingsdk/test/target-sim/__snapshots__/api.test.ts.snap]: I wonder what's causing this snapshot change

File: examples/tests/sdk_tests/bucket/events.w [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    inflight testAssertion ()  {
+        let var i = 0;
+        // waiting for up to 2 minutes
+        while i<12 {
+        i = i + 1;
+        if this.assertion() {
+          assert(this.assertion());
+          return;
+        } 
+        this.sleep(1000 * 10);
```
Comment (Chriscbr) [inline examples/tests/sdk_tests/bucket/events.w:47]: Can we make this a static method?

File: examples/tests/sdk_tests/bucket/events.w [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+    extern "../external/sleep.js" inflight sleep(ms: num);
+
+    inflight assertion (): bool {
+        return counter.peek() == this.counterVal;
+    }
+
+    inflight testAssertion ()  {
+        let var i = 0;
+        // waiting for up to 2 minutes
```
Comment (Chriscbr) [inline examples/tests/sdk_tests/bucket/events.w:40]:
```suggestion
        // waiting for up to 2 minutes, checking every 10 seconds
```

File: examples/tests/sdk_tests/bucket/events.w (hunk 4 above)
Comment (tsuf239) [reply examples/tests/sdk_tests/bucket/events.w]: done :)

File: libs/wingsdk/test/target-sim/__snapshots__/api.test.ts.snap (hunk 3 above)
Comment (tsuf239) [reply libs/wingsdk/test/target-sim/__snapshots__/api.test.ts.snap]: It happens almost every time I run it... I guess my local env is a bit different somehow...
Should I open an issue? 
@MarkMcCulloh do you have an idea what can cause it?

Comment (MarkMcCulloh) [reply libs/wingsdk/test/target-sim/__snapshots__/api.test.ts.snap]: Not sure. `express` is adding this header and I'm not sure why it's choosing one or the other.

We could explicitly configure the keepAliveTimeout setting on the express server to make sure this doesn't change.

**Pull request merged**